build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const babelify = require('babelify');
+const karmaConfig = require('../karma.conf');
+const log = require('fancy-log');
+const {
+  createKarmaServer,
+  getAdTypes,
+  runTestInBatches,
+  startTestServer,
+} = require('./helpers');
+const {createCtrlcHandler, exitCtrlcHandler} = require('../../ctrlcHandler');
+const {green, yellow, cyan, red} = require('ansi-colors');
+const {isTravisBuild} = require('../../travis');
+const {reportTestStarted} = require('.././report-test-status');
+
+class RuntimeTestConfig {
+  constructor(testType) {
+    this.testType = testType;
+
+    Object.assign(this, karmaConfig);
```

Comment:
as much as i love the brevity and terseness, I'd like to ideally know what we're assigning here to the instance. is there any way to do so?

Reply:
actually nevermind, it'd be the karma spec really.